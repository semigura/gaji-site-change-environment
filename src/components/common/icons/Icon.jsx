import classNames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';

import css from './Icon.module.scss';

const Icon = props => {
  const { className, name, ...attr } = props;
  const classes = classNames(
    css.Icon,
    className,
  );

  return (
    <svg
      // eslint-disable-next-line
      {...attr}
      role="img"
      className={classes}
    >
      <use xlinkHref={`#${name}`} />
    </svg>
  );
};

Icon.propTypes = {
  className: PropTypes.string,
  name: PropTypes.string.isRequired,
  title: PropTypes.string,
}

Icon.defaultProps = {
  className: undefined,
  title: undefined,
}

export default Icon;
