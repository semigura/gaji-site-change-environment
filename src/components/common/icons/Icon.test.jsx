import { shallow } from 'enzyme';
import React from 'react';

import Icon from './Icon';

describe('<Icon />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Icon name="logo" />);
    expect(wrapper).toMatchSnapshot();
  });
});
