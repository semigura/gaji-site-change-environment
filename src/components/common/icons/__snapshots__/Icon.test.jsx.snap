// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> renders correctly 1`] = `
<svg
  className="Icon"
  role="img"
>
  <use
    xlinkHref="#logo"
  />
</svg>
`;
